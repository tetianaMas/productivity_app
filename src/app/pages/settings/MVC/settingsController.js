import { router } from '../../../routes';
import { eventBus } from '../../../services/eventBus';

/**
 * @description controls setting's component
 * @exports SettingsController
 * @class SettingsController
 */
export class SettingsController {
  /**
   * @description Creates an instance of SettingsController
   * @param {object} view instance of SettingsView
   * @param {object} model instance of SettingsModel
   * @memberof SettingsController
   */
  constructor(view, model) {
    this.view = view;
    this.model = model;

    eventBus.subscribe('load-page', () => {
      const currentRoute = window.location.hash;

      if (/^#\/settings\/pomodoros/.test(currentRoute)) {
        this.model.getAndSaveDataFromStorage();
        this.view.renderMainPage(this.model.settingsData());
        eventBus.post('pageLoaded');
      }
    });

    eventBus.subscribe('render-dynamic-elements', () => {
      this.view.renderDynamicElements(this.model.settingsData());
      eventBus.post('pageLoaded');
    });

    eventBus.subscribe('settings-page-category-loading', () => {
      this.view.renderCategoryPage();
      eventBus.post('pageLoaded');
    });

    eventBus.subscribe('save-data', async key => {
      try {
        await this.model.dataService.sendData(
          key,
          this.model.dataService.getDataFromStorage(key)
        );

        $(document).notification('clean');
        $(document).notification({
          type: 'success',
          text: 'Settings was successfully saved',
          showTime: 3,
        });

        router.navigate('/task-list');
      } catch (error) {
        $(document).notification('clean');
        $(document).notification({
          type: 'error',
          text: 'Unable to save settings. Try again later: ' + error.message,
          showTime: 3,
        });
      }
    });

    eventBus.subscribe('increaseValue', args => this.model.increaseValue(args));
    eventBus.subscribe('decreaseValue', args => this.model.decreaseValue(args));
  }
}
