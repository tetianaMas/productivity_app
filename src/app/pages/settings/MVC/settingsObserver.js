/**
 * @description manages custom events for settings component
 * @exports SettingsObserver
 * @class SettingsObserver
 */
export class SettingsObserver {
  /**
   * @description Creates an instance of SettingsObserver
   * @memberof SettingsObserver
   */
  constructor() {
    this.observers = [];
  }

  /**
   * @description adds new callback to the observers
   * @param {object} handler callback to hangle event
   * @memberof SettingsObserver
   */
  subscribe(handler) {
    this.observers.push(handler);
  }

  /**
   * @description emits all handlers
   * @param {object} data arguments for the handler
   * @memberof SettingsObserver
   */
  notify(data) {
    this.observers.forEach(handler => handler(data));
  }
}
