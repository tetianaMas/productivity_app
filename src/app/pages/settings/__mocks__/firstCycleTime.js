export const firstCycleTime = {
  full: 175,
  hours: 2,
  minutes: 55,
};
