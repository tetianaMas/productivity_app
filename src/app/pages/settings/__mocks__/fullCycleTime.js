export const fullCycleTime = {
  full: 320,
  hours: 5,
  minutes: 20,
};
