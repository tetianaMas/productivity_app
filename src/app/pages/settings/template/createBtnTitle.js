/**
 * @description creates title for button
 * @exports createBtnTitle
 * @param {string} key current button id
 * @return {string} button title
 */
export default function (key) {
  switch (key) {
    case 'work':
      return 'work time';

    case 'iteration':
      return 'work iteration';

    case 'shortBreak':
      return 'short break';

    case 'longBreak':
      return 'long break';

    default:
      return '';
  }
}
