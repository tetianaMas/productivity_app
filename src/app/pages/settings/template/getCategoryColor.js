/**
 * @description creates category css class name for button
 * @exports getCategoryColor
 * @param {string} key current button id
 * @return {string} button category for css class
 */
export default function (key) {
  switch (key) {
    case 'work':
      return 'work';

    case 'iteration':
      return 'other';

    case 'shortBreak':
      return 'education';

    case 'longBreak':
      return 'hobby';

    default:
      return '';
  }
}
