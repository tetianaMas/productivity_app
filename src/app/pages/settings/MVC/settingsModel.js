import { Utils } from '../../../helpers/utils';
import { SettingsObserver } from './settingsObserver';

/**
 * @description manages settings' data
 * @exports SettingsModel
 * @class SettingsModel
 */
export class SettingsModel {
  /**
   * @description Creates an instance of SettingsModel
   * @param {object} dataService instance of SettingsModel
   * @memberof SettingsModel
   */
  constructor(dataService) {
    this.dataService = dataService;
    this.buttonsValuesChangedEvent = new SettingsObserver();
    this.buttonsValues = null;
    this.cycleCount = 2;
    this.minutesInSchedule = 30;
    this.settingsData = () => this.getAllSettingsData();
  }

  /**
   * @description gets and saves data
   * @memberof SettingsModel
   */
  getAndSaveDataFromStorage() {
    const dataFromStorage = JSON.parse(sessionStorage.getItem('settings'));
    this.buttonsValues = Utils.sortObjectByKeys(dataFromStorage);
  }

  /**
   * @description gets data for current settings' shedule state
   * @return {object} data for rendering schedule
   * @memberof SettingsModel
   */
  getAllSettingsData() {
    const settingsData = {
      buttonsValues: this.buttonsValues,
      cycleCount: this.cycleCount,
      minutesInSchedule: this.minutesInSchedule,
      fullCycleTime: this.getFullCycleTime(),
      firstCycleTime: this.getFirstCycleTime(),
      secondCycleTime:
        this.getFullCycleTime().full - this.getFirstCycleTime().full,
      timeScheduleDelimeter:
        this.getFullCycleTime().full % this.minutesInSchedule,
    };

    return settingsData;
  }

  /**
   * @description saves data and emits 'buttonsValuesChangedEvent' event
   * @param {object} data settings' data
   * @memberof SettingsModel
   */
  saveSettingsData(data) {
    sessionStorage.setItem('settings', JSON.stringify(data));
    this.buttonsValuesChangedEvent.notify(this.settingsData());
  }

  /**
   * @description calculates time for settings' shedule
   * @return {object} full time for shedule
   * @memberof SettingsModel
   */
  getFullCycleTime() {
    const MINUTES_IN_HOUR = 60;
    const data = this.buttonsValues;

    const fullTime =
      (data.work.value * data.iteration.value +
        data.shortBreak.value * (data.iteration.value - 1)) *
        this.cycleCount +
      data.longBreak.value * (this.cycleCount - 1);

    const fullTimeHours = Math.floor(fullTime / MINUTES_IN_HOUR);
    const fullTimeMinutes = Math.floor(fullTime % MINUTES_IN_HOUR);

    return {
      full: fullTime,
      hours: fullTimeHours,
      minutes: fullTimeMinutes,
    };
  }

  /**
   * @description calculates time for settings' first cycle shedule
   * @return {object} first Cycle Time for shedule
   * @memberof SettingsModel
   */
  getFirstCycleTime() {
    const MINUTES_IN_HOUR = 60;
    const data = this.buttonsValues;

    const firstCycleTime =
      data.work.value * data.iteration.value +
      data.shortBreak.value * (data.iteration.value - 1) +
      data.longBreak.value;

    const firstCycleTimeHours = Math.floor(firstCycleTime / MINUTES_IN_HOUR);
    const firstCycleTimeMinutes = Math.floor(firstCycleTime % MINUTES_IN_HOUR);

    return {
      full: firstCycleTime,
      hours: firstCycleTimeHours,
      minutes: firstCycleTimeMinutes,
    };
  }

  /**
   * @description increases button's value
   * @param {object} args id of button and current button's value
   * @memberof SettingsModel
   */
  increaseValue(args) {
    if (!Array.isArray(args)) return;
    const [id, currentValue] = args;
    if (
      !id ||
      typeof id !== 'string' ||
      !currentValue ||
      typeof currentValue !== 'number'
    ) {
      return;
    }
    const newValue = currentValue + this.buttonsValues[id].step;
    this.setValue(id, newValue);
  }

  /**
   * @description decreases button's value
   * @param {object} args id of button and current button's value
   * @memberof SettingsModel
   */
  decreaseValue(args) {
    if (!Array.isArray(args)) return;
    const [id, currentValue] = args;
    if (
      !id ||
      typeof id !== 'string' ||
      !currentValue ||
      typeof currentValue !== 'number'
    ) {
      return;
    }
    const newValue = currentValue - this.buttonsValues[id].step;
    this.setValue(id, newValue);
  }
  /**
   * @description sets button's value
   * @param {number} id button's id
   * @param {number} newValue new button's value
   * @memberof SettingsModel
   */
  setValue(id, newValue) {
    if (typeof id !== 'string' || typeof newValue !== 'number') return;
    if (
      newValue >= this.buttonsValues[id].min &&
      newValue <= this.buttonsValues[id].max
    ) {
      this.buttonsValues[id].value = newValue;
      this.saveSettingsData(this.buttonsValues);
    }
  }

  /**
   * @description sets default buttons' values
   * @memberof SettingsModel
   */
  setDefaultButtonsValues() {
    for (const [key] of Object.entries(this.buttonsValues)) {
      this.setValue(key, this.buttonsValues[key].default);
    }
  }
}
