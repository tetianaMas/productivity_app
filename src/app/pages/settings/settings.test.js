/* eslint-disable no-undef */
import { settingsData } from './__mocks__/settingsData';
import { buttonsValues } from './__mocks__/buttonsValues';
import buttonsTemplate from './template/settings-buttons.hbs';
import settingsTemplate from './template/settings.hbs';
import categoriesTemplate from './template/categories.hbs';

import { SettingsModel } from './MVC/settingsModel';
import { dataService } from '../../services/dataService/dataService';
import { allSettingsData } from './__mocks__/allSettingsData';
import { fullCycleTime } from './__mocks__/fullCycleTime';
import { firstCycleTime } from './__mocks__/firstCycleTime';
import createBtnTitle from './template/createBtnTitle';
import getCategoryColor from './template/getCategoryColor';
import getCurrentDimention from './template/getCurrentDimension';

const settingsModel = new SettingsModel(dataService);

describe('rendering', () => {
  test('renders correctly page', () => {
    const tree = JSON.stringify(settingsTemplate(settingsData));
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly buttons', () => {
    const tree = JSON.stringify(buttonsTemplate({ values: buttonsValues }));
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly categories', () => {
    const tree = JSON.stringify(
      categoriesTemplate({
        tab: {
          firstVal: 'pomodoros',
          secondVal: 'categories',
          activeTab: true,
          firstValAttr: 'data-tab="pomodoros"',
          secondValAttr: 'data-tab="categories"',
        },
      })
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('createBtnTitle', () => {
  test('should return right title', () => {
    const res = createBtnTitle('work');
    expect(res).toBe('work time');
  });

  test('should return right title', () => {
    const res = createBtnTitle('iteration');
    expect(res).toBe('work iteration');
  });

  test('should return right title', () => {
    const res = createBtnTitle('shortBreak');
    expect(res).toBe('short break');
  });

  test('should return right title', () => {
    const res = createBtnTitle('longBreak');
    expect(res).toBe('long break');
  });

  test('should return falsy value', () => {
    const res = createBtnTitle('some wrong text');
    expect(res).toBeFalsy();
  });

  test('should return falsy value', () => {
    const res = createBtnTitle(345353464365);
    expect(res).toBeFalsy();
  });

  test('should return falsy value', () => {
    const res = createBtnTitle();
    expect(res).toBeFalsy();
  });
});

describe('getCategoryColor', () => {
  test('should return right category color', () => {
    const res = getCategoryColor('work');
    expect(res).toBe('work');
  });

  test('should return right category color', () => {
    const res = getCategoryColor('iteration');
    expect(res).toBe('other');
  });

  test('should return right category color', () => {
    const res = getCategoryColor('shortBreak');
    expect(res).toBe('education');
  });

  test('should return right category color', () => {
    const res = getCategoryColor('longBreak');
    expect(res).toBe('hobby');
  });

  test('should return falsy value', () => {
    const res = getCategoryColor('some wrong text');
    expect(res).toBeFalsy();
  });

  test('should return falsy value', () => {
    const res = getCategoryColor(345353464365);
    expect(res).toBeFalsy();
  });

  test('should return falsy value', () => {
    const res = getCategoryColor();
    expect(res).toBeFalsy();
  });
});

describe('getCurrentDimention', () => {
  test('should return right dimention', () => {
    const res = getCurrentDimention('iteration');
    expect(res).toBe('iterations');
  });

  test('should return `minutes`', () => {
    const res = getCurrentDimention('some wrong text');
    expect(res).toBe('minutes');
  });

  test('should return `minutes`', () => {
    const res = getCurrentDimention(345353464365);
    expect(res).toBe('minutes');
  });

  test('should return `minutes`', () => {
    const res = getCurrentDimention();
    expect(res).toBe('minutes');
  });
});

describe('getting settings data', () => {
  beforeEach(() => {
    settingsModel.buttonsValues = buttonsValues;
  });
  test('should return settings data', () => {
    const result = settingsModel.getAllSettingsData();
    expect(result).toEqual(allSettingsData);
  });

  test('should return full cycle time data', () => {
    const result = settingsModel.getFullCycleTime();
    expect(result).toEqual(fullCycleTime);
  });

  test('should return first cycle time data', () => {
    const result = settingsModel.getFirstCycleTime();
    expect(result).toEqual(firstCycleTime);
  });
});

describe('increasing/decreasing values', () => {
  beforeEach(() => {
    settingsModel.buttonsValues = buttonsValues;
    spyOn(settingsModel, 'setValue');
  });

  test('should call set button value func', () => {
    settingsModel.increaseValue(['longBreak', 20]);
    expect(settingsModel.setValue).toBeCalled();
    expect(settingsModel.setValue).toHaveBeenCalledTimes(1);
  });

  test('should call set button value func', () => {
    settingsModel.decreaseValue(['shortBreak', 5]);
    expect(settingsModel.setValue).toBeCalled();
    expect(settingsModel.setValue).toHaveBeenCalledTimes(1);
  });

  test('should not call set button value func', () => {
    const result = settingsModel.decreaseValue([3456, 5]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  test('should not call set button value func', () => {
    const result = settingsModel.decreaseValue([true, false]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  test('should not call set button value func', () => {
    const result = settingsModel.decreaseValue();
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  test('should not call set button value func', () => {
    const result = settingsModel.increaseValue([3456, 5]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  test('should not call set button value func', () => {
    const result = settingsModel.increaseValue([true, false]);
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });

  test('should not call set button value func', () => {
    const result = settingsModel.increaseValue();
    expect(result).toBeFalsy();
    expect(settingsModel.setValue).not.toBeCalled();
  });
});

describe('setting values', () => {
  beforeEach(() => {
    settingsModel.buttonsValues = buttonsValues;
  });

  test('should set button value', () => {
    const arg = ['work', 25];
    const [id, currentValue] = arg;
    const newValue = currentValue - settingsModel.buttonsValues[id].step;

    settingsModel.setValue(id, newValue);
    expect(settingsModel.buttonsValues[id].value).toBe(newValue);
  });

  test('should not set button value', () => {
    settingsModel.setValue('work', 45);
    expect(settingsModel.buttonsValues['work'].value).toBe(20);
  });

  test('should return falsy value', () => {
    const res = settingsModel.setValue(34, 56);
    expect(res).toBeFalsy();
  });

  test('should return falsy value', () => {
    const res = settingsModel.setValue();
    expect(res).toBeFalsy();
  });
});
