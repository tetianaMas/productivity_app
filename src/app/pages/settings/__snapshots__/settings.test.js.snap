// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly buttons 1`] = `"\\"<div class=\\\\\\"settings__item settings__item--other\\\\\\">\\\\r\\\\n  <div class=\\\\\\"settings__info\\\\\\">\\\\r\\\\n    <span>\\\\r\\\\n      work iteration\\\\r\\\\n    </span>\\\\r\\\\n    <div class=\\\\\\"settings__controls\\\\\\" id=\\\\\\"iteration\\\\\\">\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-minus js-btn js-btn-minus\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n      <span class=\\\\\\"settings__controls__value js-btn-value\\\\\\">\\\\r\\\\n        5\\\\r\\\\n      </span>\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-add js-btn js-btn-add\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <p class=\\\\\\"settings__item__caption\\\\\\">\\\\r\\\\n    Please select a value between 2 and 5\\\\r\\\\n    <span class=\\\\\\"settings__item__caption--white\\\\\\">\\\\r\\\\n      iterations\\\\r\\\\n    </span>\\\\r\\\\n  </p>\\\\r\\\\n</div>\\\\r\\\\n<div class=\\\\\\"settings__item settings__item--hobby\\\\\\">\\\\r\\\\n  <div class=\\\\\\"settings__info\\\\\\">\\\\r\\\\n    <span>\\\\r\\\\n      long break\\\\r\\\\n    </span>\\\\r\\\\n    <div class=\\\\\\"settings__controls\\\\\\" id=\\\\\\"longBreak\\\\\\">\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-minus js-btn js-btn-minus\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n      <span class=\\\\\\"settings__controls__value js-btn-value\\\\\\">\\\\r\\\\n        30\\\\r\\\\n      </span>\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-add js-btn js-btn-add\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <p class=\\\\\\"settings__item__caption\\\\\\">\\\\r\\\\n    Please select a value between 15 and 30\\\\r\\\\n    <span class=\\\\\\"settings__item__caption--white\\\\\\">\\\\r\\\\n      minutes\\\\r\\\\n    </span>\\\\r\\\\n  </p>\\\\r\\\\n</div>\\\\r\\\\n<div class=\\\\\\"settings__item settings__item--education\\\\\\">\\\\r\\\\n  <div class=\\\\\\"settings__info\\\\\\">\\\\r\\\\n    <span>\\\\r\\\\n      short break\\\\r\\\\n    </span>\\\\r\\\\n    <div class=\\\\\\"settings__controls\\\\\\" id=\\\\\\"shortBreak\\\\\\">\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-minus js-btn js-btn-minus\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n      <span class=\\\\\\"settings__controls__value js-btn-value\\\\\\">\\\\r\\\\n        5\\\\r\\\\n      </span>\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-add js-btn js-btn-add\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <p class=\\\\\\"settings__item__caption\\\\\\">\\\\r\\\\n    Please select a value between 3 and 5\\\\r\\\\n    <span class=\\\\\\"settings__item__caption--white\\\\\\">\\\\r\\\\n      minutes\\\\r\\\\n    </span>\\\\r\\\\n  </p>\\\\r\\\\n</div>\\\\r\\\\n<div class=\\\\\\"settings__item settings__item--work\\\\\\">\\\\r\\\\n  <div class=\\\\\\"settings__info\\\\\\">\\\\r\\\\n    <span>\\\\r\\\\n      work time\\\\r\\\\n    </span>\\\\r\\\\n    <div class=\\\\\\"settings__controls\\\\\\" id=\\\\\\"work\\\\\\">\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-minus js-btn js-btn-minus\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n      <span class=\\\\\\"settings__controls__value js-btn-value\\\\\\">\\\\r\\\\n        25\\\\r\\\\n      </span>\\\\r\\\\n      <button\\\\r\\\\n        class=\\\\\\"settings__controls__btn icon-add js-btn js-btn-add\\\\\\"\\\\r\\\\n      ></button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <p class=\\\\\\"settings__item__caption\\\\\\">\\\\r\\\\n    Please select a value between 15 and 25\\\\r\\\\n    <span class=\\\\\\"settings__item__caption--white\\\\\\">\\\\r\\\\n      minutes\\\\r\\\\n    </span>\\\\r\\\\n  </p>\\\\r\\\\n</div>\\\\r\\\\n\\""`;

exports[`rendering renders correctly categories 1`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header><main class=\\\\\\"main\\\\\\">\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right \\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab tab__btn--active \\\\\\" data-tab=\\\\\\"pomodoros\\\\\\">pomodoros</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-tab=\\\\\\"categories\\\\\\">categories</button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <section class=\\\\\\"main__content main__content--categories\\\\\\" aria-label=\\\\\\"Categories List\\\\\\">\\\\r\\\\n      <h2 class=\\\\\\"main__content__title\\\\\\">Categories list overview</h2>\\\\r\\\\n\\\\r\\\\n      <ul class=\\\\\\"list\\\\\\">\\\\r\\\\n        <li class=\\\\\\"list__item settings__item--work\\\\\\">work</li>\\\\r\\\\n        <li class=\\\\\\"list__item settings__item--education\\\\\\">education</li>\\\\r\\\\n        <li class=\\\\\\"list__item settings__item--hobby\\\\\\">hobby</li>\\\\r\\\\n        <li class=\\\\\\"list__item settings__item--sport\\\\\\">sport</li>\\\\r\\\\n        <li class=\\\\\\"list__item settings__item--other\\\\\\">other</li>\\\\r\\\\n      </ul>\\\\r\\\\n\\\\r\\\\n      <div class=\\\\\\"main__control\\\\\\">\\\\r\\\\n        <button class=\\\\\\"main__control__button main__control__button--blue js-btn-to-tasks\\\\\\">\\\\r\\\\n          Go to Tasks\\\\r\\\\n        </button>\\\\r\\\\n      </div>\\\\r\\\\n    </section>\\\\r\\\\n</main>\\\\r\\\\n\\\\r\\\\n\\""`;

exports[`rendering renders correctly page 1`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header><main class=\\\\\\"main\\\\\\">\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right \\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab tab__btn--active \\\\\\" data-tab=\\\\\\"pomodoros\\\\\\">pomodoros</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-tab=\\\\\\"categories\\\\\\">categories</button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <section class=\\\\\\"main__content main__content--pomodoros\\\\\\" aria-label=\\\\\\"Pomodoros List\\\\\\">\\\\r\\\\n        <h2 class=\\\\\\"main__content__title\\\\\\">Pomodoros settings</h2>\\\\r\\\\n\\\\r\\\\n        <div class=\\\\\\"settings js-settings\\\\\\"></div>\\\\r\\\\n\\\\r\\\\n        <div class=\\\\\\"cycle\\\\\\">\\\\r\\\\n          <h2 class=\\\\\\"cycle__title\\\\\\">Your cycle</h2>\\\\r\\\\n          <section class=\\\\\\"cycle__scale js-cycle\\\\\\" aria-label=\\\\\\"Dinamic cycle represents time settings for pomodoros\\\\\\">\\\\r\\\\n          </section>\\\\r\\\\n          <div class=\\\\\\"cycle__control\\\\\\">\\\\r\\\\n            <button class=\\\\\\"cycle__button cycle__button--blue js-btn-to-tasks\\\\\\">\\\\r\\\\n              Go to Tasks\\\\r\\\\n            </button>\\\\r\\\\n            <button class=\\\\\\"cycle__button cycle__button--mint js-btn-save\\\\\\">\\\\r\\\\n              Save\\\\r\\\\n            </button>\\\\r\\\\n          </div>\\\\r\\\\n        </div>\\\\r\\\\n    </section>\\\\r\\\\n</main>\\\\r\\\\n\\\\r\\\\n\\""`;
